tour_manager/src/main/java/edu/artic/tours/manager/TourProgressManager.kt

```diff
@@ -23,4 +27,57 @@ class TourProgressManager {
     val selectedTour: Subject<Optional<ArticTour>> = BehaviorSubject.createDefault(Optional(null))
     val proposedTour: Subject<Optional<Pair<ArticTour, ArticTour.TourStop>>> = BehaviorSubject.createDefault(Optional(null))
     val leaveTourRequest: Subject<Boolean> = PublishSubject.create()
+
+    /**
+     * If [audioFileModel] belongs to currently selected tour i.e. [selectedTour],
+     * this method advances [selectedStop] to next tour stop.
+     */
+    @SuppressLint("CheckResult")
+    fun playBackEnded(audioFileModel: AudioFileModel) {
+        Observables.combineLatest(selectedTour.filterValue(), selectedStop)
+                .take(1)
+                .subscribe { (tour, currentStopID) ->
+
+                    /**
+                     * Get audio file id
+                     */
+                    val audioFileID: String? = if (currentStopID == INTRO_TOUR_STOP_OBJECT_ID) {
+                        tour.tourAudio
+                    } else {
+                        tour.tourStops.find { it.objectId == currentStopID }?.audioId
+                    }
+
+                    audioFileID?.let { audioID ->
+                        audioFileDao
+                                .getAudioByIdAsync(audioID)
+                                .toObservable()
+                                .subscribe { audioFile ->
```

Comment:
Please add a simple `onError` to this subscription, on the off chance that `audioID` is malformed. Something that does nothing in production but uses Timber in debug, perhaps?

Reply:
Added.